test/e2e/subscription_e2e_test.go

```diff
@@ -1366,6 +1366,258 @@ var _ = Describe("Subscription", func() {
+				Expect(subscription).ToNot(BeNil())
+
+				_, err = fetchCSV(crClient, mainCSVName, testNamespace, csvSucceededChecker)
+				Expect(err).ToNot(HaveOccurred())
+
+				// ensure correct CSVs were picked
+				Eventually(func() ([]string, error) {
+					ip, err := crClient.OperatorsV1alpha1().InstallPlans(testNamespace).Get(context.TODO(), subscription.Status.InstallPlanRef.Name, metav1.GetOptions{})
+					if err != nil {
+						return nil, err
+					}
+					return ip.Spec.ClusterServiceVersionNames, nil
+				}).Should(ConsistOf(mainCSVName, rightCSVName))
+			})
+		}
+
+		It("the same dependency exists on catsrcs with different name", func() {
+			// Ensure resolver is deterministic.
+			packageDepRight = registry.PackageManifest{PackageName: "PackageDependent"}
+			csv := newCSV(rightCSVName, testNamespace, "", semver.MustParse("0.1.0"),
+				[]apiextensions.CustomResourceDefinition{crd}, nil, nil)
+			packageDepRight.DefaultChannelName = alphaChannel
+			packageDepRight.Channels = []registry.PackageChannel{{Name: alphaChannel, CurrentCSVName: csv.GetName()}}
+			csvsRight = append(csvsRight, csv)
+
+			csv.Name = wrongCSVName
+			packageDepWrong = packageDepRight
+			packageDepWrong.Channels = []registry.PackageChannel{{Name: alphaChannel, CurrentCSVName: csv.GetName()}}
+			csvsWrong = append(csvsWrong, csv)
+
+			By("creating catalogsources", func() {
+				var c1, c2, c3 cleanupFunc
+				catsrcMain, c1 = createInternalCatalogSource(kubeClient, crClient, genName("catsrc"), testNamespace,
+					[]registry.PackageManifest{packageMain}, nil, csvsMain)
+
+				catsrcDepRight, c2 = createInternalCatalogSource(kubeClient, crClient, "catsrc1", testNamespace,
+					[]registry.PackageManifest{packageDepRight}, []apiextensions.CustomResourceDefinition{crd}, csvsRight)
+				catsrcDepRight.Spec.Priority = 100
+
+				catsrcDepWrong, c3 = createInternalCatalogSource(kubeClient, crClient, "catsrc2", testNamespace,
+					[]registry.PackageManifest{packageDepWrong}, []apiextensions.CustomResourceDefinition{crd}, csvsWrong)
+				catsrcDepWrong.Spec.Priority = 100
+
+				cleanup = func() {
+					c1()
+					c2()
+					c3()
+				}
+			})
+
+			By("waiting for catalogsources to be ready", func() {
+				_, err := fetchCatalogSourceOnStatus(crClient, catsrcMain.GetName(), testNamespace, catalogSourceRegistryPodSynced)
+				Expect(err).ToNot(HaveOccurred())
+				_, err = fetchCatalogSourceOnStatus(crClient, catsrcDepRight.GetName(), testNamespace, catalogSourceRegistryPodSynced)
+				Expect(err).ToNot(HaveOccurred())
+				_, err = fetchCatalogSourceOnStatus(crClient, catsrcDepWrong.GetName(), testNamespace, catalogSourceRegistryPodSynced)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			By("choose the dependent package from the same catsrc that has a higher priority value", func() {
+				ensureInstalledCSVs()
+			})
+		})
+
+		It("the same dependency exists on both the catsrc of the installing operator and another catsrc with a higher"+
+			" priority", func() {
```

Comment:
Nit: Splitting the spec description across two lines is funky. There's no hard line length rule, and if there were, this would probably be an exception to it.

Reply:
lol, goland, i sorta need to disable that character count function.

Reply:
Haha, it was only trying to help! :pencil2: